spec/ruby/core/env/delete_spec.rb

```diff
@@ -16,9 +16,25 @@
     ENV.delete("foo").should == "bar"
   end
 
+  it "ignores the block if the environment variable exists" do
+    ENV["foo"] = "bar"
+    begin
+      -> { ENV.delete("foo") { |name| fail name } }.should_not raise_error(RuntimeError)
+    end
+  end
+
   it "yields the name to the given block if the named environment variable does not exist" do
     ENV.delete("foo")
     ENV.delete("foo") { |name| ScratchPad.record name }
     ScratchPad.recorded.should == "foo"
   end
+
+  it "returns nil if the named environment variable does not exist and block given" do
```

Comment:
This spec expects the previous, buggy behavior.  The fix in hash.c breaks that old spec, which needs to be updated to reflect the fixed behavior.  The spec should expect the return value to be the return value of the block, not nil.